group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilder.java

```diff
@@ -293,14 +293,18 @@ public TargetAssignmentBuilder removeMember(
      * @throws PartitionAssignorException if the target assignment cannot be computed.
      */
     public TargetAssignmentResult build() throws PartitionAssignorException {
-        Map<String, AssignmentMemberSpec> memberSpecs = new HashMap<>();
+        Map<String, MemberSubscriptionSpecImpl> memberSpecs = new HashMap<>(members.size());
 
         // Prepare the member spec for all members.
-        members.forEach((memberId, member) -> memberSpecs.put(memberId, createAssignmentMemberSpec(
-            member,
-            targetAssignment.getOrDefault(memberId, Assignment.EMPTY),
-            topicsImage
-        )));
+        members.forEach((memberId, member) -> {
+            Assignment assignment = targetAssignment.getOrDefault(memberId, Assignment.EMPTY);
+
```

Comment:
nit: We could inline `targetAssignment.getOrDefault(memberId, Assignment.EMPTY)` as it was before.